### PR TITLE
OCPBUGS-45022: Obfuscate sensitive data in Proxy resource while inspection

### PR DESCRIPTION
**Pull request by ardaguclu**
Proxy resource can contain sensitive information and inspect command should obfuscate all these fields. This PR achieves this.

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
until pre-merge tests are completed.

**Comment by ardaguclu** [comment]
unrelated;
/test e2e-agnostic-ovn-cmd

**Comment by zhouying7780** [comment]
@ardaguclu  I tested , I hit error with the pr, so we should have triggered new issue with the pr, please help take a look for the comments in the bug.

**Comment by ardaguclu** [comment]
@ingvagabund ptal, when you are available?. Thanks.

File: pkg/cli/admin/inspect/proxy.go
Comment (ingvagabund) [inline pkg/cli/admin/inspect/proxy.go:20]: Is this method used anywhere? Is there an interface proxyList needs to implement?

File: pkg/cli/admin/inspect/proxy.go
Comment (ingvagabund) [inline pkg/cli/admin/inspect/proxy.go:56]: Can you also unit test `elideProxy` or `inspectProxyInfo`?

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Checking https://github.com/openshift/oc/tree/master/pkg/cli/admin/inspect there are no unit tests yet. Though, there are other `elideXXX` functions that are worth unit testing as well.

File: pkg/cli/admin/inspect/proxy.go
Comment (ardaguclu) [reply pkg/cli/admin/inspect/proxy.go:56]: Added.

File: pkg/cli/admin/inspect/proxy.go
Comment (ardaguclu) [reply pkg/cli/admin/inspect/proxy.go:20]: It needs the implementation of this function to satisfy `listAccessor` interface in https://github.com/openshift/oc/blob/89781e0b8f29052e3350b74b677f3c69eea096f1/pkg/cli/admin/inspect/resource.go#L128 to be populated into the list https://github.com/openshift/oc/blob/89781e0b8f29052e3350b74b677f3c69eea096f1/pkg/cli/admin/inspect/resource.go#L179

Comment (ingvagabund) [reply pkg/cli/admin/inspect/proxy.go:20]: `var _ listAccessor = &proxyList{}` will do then

Comment (ardaguclu) [reply pkg/cli/admin/inspect/proxy.go:22]: This https://github.com/openshift/oc/pull/1929/files#diff-6f6b5698b5c7ebfc01c64967180a4d2e5f2f248509dae99f7d0d07e13018c366R151-R153 ensures that `proxyList` implements `listAccessor` but in any case, I've added your suggestion above.

**Pull request merged**
